scipy/sparse/linalg/dsolve/linsolve.py

```diff
@@ -79,7 +79,16 @@ def _get_umf_family(A):
             % (f_type, i_type)
         raise ValueError(msg)
 
-    return family
+    if np.prod(A.shape) > np.iinfo(np.int32).max:
```

Comment:
```suggestion
    if np.prod(A.shape.astype(np.int64)) > np.iinfo(np.int32).max:
```
Or should we convert each element to Python integers to preclude overflow entirely?